eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubSharedAccessKeyCredential.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.eventhubs;
+
+import com.azure.core.credentials.TokenCredential;
+import com.azure.core.implementation.util.ImplUtils;
+import reactor.core.publisher.Mono;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Base64;
+import java.util.Locale;
+import java.util.Objects;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Authorizes with Azure Event Hubs service using a shared access key from either an Event Hubs namespace or a specific
+ * Event Hub.
+ */
+public class EventHubSharedAccessKeyCredential extends TokenCredential {
+    private static final String SHARED_ACCESS_SIGNATURE_FORMAT = "SharedAccessSignature sr=%s&sig=%s&se=%s&skn=%s";
+    private static final String HASH_ALGORITHM = "HMACSHA256";
+
+    private final String keyName;
+    private final Mac hmac;
+    private final Duration tokenValidity;
+
+    /**
+     * Creates an instance that authorizes using the {@code keyName} and {@code sharedAccessKey}. The authorization
+     * lasts for a period of {@code tokenValidity} before another token must be requested.
+     *
+     * @param keyName Name of the shared access key policy.
+     * @param sharedAccessKey Value of the shared access key.
+     * @param tokenValidity The duration for which the shared access signature is valid.
+     * @throws IllegalArgumentException if {@code keyName}, {@code sharedAccessKey} is null or empty. Or the value of
+     * {@code tokenValidity} less less than 1.
```

Comment:
two comments:
1. "less less"
2. What does "less than 1" mean in the context of a Duration? I'm sure you mean it has to be a positive value, but the way it reads I could argue you're saying less than one year / one month / one millisecond are all equally invalid.